.circleci/config.yml

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2016-2018 The OpenZipkin Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:9-jdk
+        environment:
+          # Quiet Maven invoker logs (Downloading... when running /src/it)
+          MAVEN_OPTS: -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+
+      - image: circleci/mysql:5.7
+        environment:
+          MYSQL_USER: root
+
+    steps:
+      - checkout
+      - restore_cache:
+          key: offline-dependencies-{{ checksum "pom.xml" }}
+
+      - run:
+          name: Download dependencies
+          command: |
+            sudo apt-get install xsltproc
+            ./build-support/go-offline.sh
+
+      - save_cache:
+          key: offline-dependencies-{{ checksum "pom.xml" }}
+          paths:
+            - ~/.m2
+
+      - run:
+          name: Tests
+          command: |
+            # skip license on circleci due to #1512
+            ./mvnw verify -Dlicense.skip=true
+
+      - run:
+          name: Collect test reports
+          when: always
+          command: |
+            mkdir -p /tmp/test-reports/unit-tests/
+            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} /tmp/test-reports/unit-tests/ \;
```

Comment:
Under the directory you use for `path`, each directory will be used as the "name" of the test results contained in files in that directory. This is why it takes the test results from surefire and puts then in a `unit-tests` directory under the `path` we point to, and it takes all the test results from failsafe and puts them in a `integration-tests` directory. So if CircleCI showed things in a more sophisticated way, you would be able to see test results distinguished as unit tests or integration tests.

As for storing the test results, I think it would be fine. I'm not sure if CircleCI does much more than allow you download the individual files for a given build, which is better than nothing but does seem cumbersome to get the details about a test failure/timing. The same information should be in the build log, so it becomes a question of whether finding it there or finding the corresponding test result file and downloading it is more useful. That is, if my assumption that CircleCI doesn't do any analysis/display of the files for us.

Reply:
> It feels like directly pointing CircleCI to the test results would be cleaner.

As I mentioned in the other issue, I agree. But when it comes to multi-module/nested projects, I don't think CircleCI (particularly 2.0) has any smarts about this. If there is just one root project, we can just target directly. With more than one, though, CircleCI 2.0 is so generic it doesn't try to understand that a multi-module build is happening and collect from each sub-project, as far as I understand.

Reply:
Gotcha, thanks for explaining. That all makes sense.